File: LeetCode Solutions/Arrays 101/Sort Array By Parity/Sort Array By Parity.java
class Solution {
    public int[] sortArrayByParity(int[] A) {
        int left = 0;
        int right = A.length - 1;
        while (left < right){
            if (A[left] % 2 == 1){
                int temp = A[left];
                A[left] = A[right];
                A[right] = temp;
                right -= 1;
            } else{
                left += 1;
            }
        }
        return A;
    }
}

File: LeetCode Solutions/May 2022 LeetCoding Challenge/Sort Array By Parity.cpp
class Solution {
public:
    vector<int> sortArrayByParity(vector<int>& nums) {
        int left = 0, right = nums.size() - 1, temp;
        while(left < right){
            if(nums[left] % 2 == 1){
                temp = nums[left];
                nums[left] = nums[right];
                nums[right] = temp;
                right -= 1;
            } else{
                left += 1;
            }
        }
        return nums;
    }
};